compiler/src/dotty/tools/dotc/fromtasty/Debug.scala

```diff
@@ -0,0 +1,60 @@
+package dotty.tools
+package dotc
+package fromtasty
+
+import scala.util.control.NonFatal
+
+import dotty.tools.io.Path
+
+import java.nio.file.{Files, Paths}
+
+object Debug {
+  def main(args: Array[String]): Unit = {
+    // Preload scala.util.control.NonFatal. Otherwise, when trying to catch a StackOverflowError,
+    // we may try to load it but fail with another StackOverflowError and lose the original exception,
+    // see <https://groups.google.com/forum/#!topic/scala-user/kte6nak-zPM>.
+    val _ = NonFatal
+
+
+    println("From tasty debug driver")
+    assert(!args.contains("-d"))
+
+    val fromSourcesOut = Files.createTempDirectory(Paths.get("out").toAbsolutePath, "from-sources-tmp")
+
+    println(s"Compiling scala to sources to $fromSourcesOut")
+    val compilation1 = dotc.Main.process("-d" +: fromSourcesOut.toString +: args)
+
+    if (compilation1.hasErrors) {
+      println("Failed compilation from sources")
+      sys.exit(1)
+    }
+
+    val fromTastyOut = Files.createTempDirectory(Paths.get("out").toAbsolutePath, "from-tasty-tmp")
```

Comment:
Why a temporary directory?

Reply:
To ensure it is a clean directory and the only contents are the ones coming from this compilation.